### PR TITLE
[MM-20469] migrate emoji_list_item to typescript

### PR DESCRIPTION
**Pull request by Tak-Iwamoto**
#### Summary
- migrate components/emoji/emoji_list_item to typescript

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/15471
JIRA: https://mattermost.atlassian.net/browse/MM-20469

### CODE REVIEW

File: components/emoji/emoji_list_item/emoji_list_item.tsx [hunk 1]
```diff
@@ -2,35 +2,28 @@
...
+    actions: {
+        deleteCustomEmoji: (id: string) => void;
     }
-
-    static defaultProps = {
-        emoji: {},
-        currentUserId: '',
-        currentTeam: {},
-        creatorDisplayName: '',
-    }
```
Comment (cpoile) [inline components/emoji/emoji_list_item/emoji_list_item.tsx:31]: Could you still include the defaultProps?

File: components/emoji/emoji_list_item/emoji_list_item.tsx [hunk 2]
```diff
@@ -57,7 +50,7 @@ export default class EmojiListItem extends React.PureComponent {
         } else {
             deleteButton = (
                 <AnyTeamPermissionGate permissions={[Permissions.DELETE_EMOJIS]}>
-                    <AnyTeamPermissionGate permissions={[Permissions.DELETE_OTHERS_EMOJIS]}>
+                    <AnyTeamPermissionGate permissions={[Permissions.DELETE_EMOJIS]}>
```
Comment (cpoile) [inline components/emoji/emoji_list_item/emoji_list_item.tsx:53]: Did you intend to change this permission?

**Review by cpoile — CHANGES_REQUESTED** [review]
Hi @Tak-Iwamoto, thank you so much for the contribution! 
I have a couple questions for you below. Thanks again!

File: components/emoji/emoji_list_item/emoji_list_item.tsx (hunk 1 above)
Comment (Tak-Iwamoto) [reply components/emoji/emoji_list_item/emoji_list_item.tsx:31]: OK

File: components/emoji/emoji_list_item/emoji_list_item.tsx (hunk 2 above)
Comment (Tak-Iwamoto) [reply components/emoji/emoji_list_item/emoji_list_item.tsx]: Oh, it is my mistake...
I put it back.

File: components/emoji/emoji_list_item/emoji_list_item.tsx [hunk 3]
```diff
@@ -2,35 +2,35 @@
...
-        }).isRequired,
+export type Props = {
+    emoji: CustomEmoji;
+    currentUserId: string;
+    creatorDisplayName: string;
+    creatorUsername?: string;
+    currentTeam?: Team;
+    onDelete?: (id: string) => void;
+    actions: {
+        deleteCustomEmoji: (id: string) => void;
```
Comment (Ayanrocks) [inline components/emoji/emoji_list_item/emoji_list_item.tsx:22]: I think this `deleteCustomEmoji` should be like this
```ts
deleteCustomEmoji: (emojiId: string) => Promise<{data: boolean, error: ServerError}>
```
https://github.com/mattermost/mattermost-redux/blob/4aa23838f96f94326bdc8b291d7674079da8bb5d/src/actions/emojis.ts#L186
@cpoile Any thoughts?

Comment (cpoile) [reply components/emoji/emoji_list_item/emoji_list_item.tsx:22]: Good catch @Ayanrocks -- but I think it needs to be:

```suggestion
        deleteCustomEmoji: (id: string) => ActionFunc;
```

**Review by cpoile — CHANGES_REQUESTED** [review]
One more suggestion @Tak-Iwamoto, thank you again for the submission!

**Comment by jfrerich** [comment]
Hello again, @Tak-Iwamoto. Thanks for all of your contributions!  It would be great to get these pushed through to completion and merged.  Do you need any help?

**Comment by Ayanrocks** [comment]
@ashishbhate @hanzei @jfrerich Can I take this PR? Can I continue in this PR or make a new one?

**Comment by hanzei** [comment]
@Ayanrocks I don't think you can push to this PR. Would you please open a new one and I will close this?

**Comment by Ayanrocks** [comment]
> @Ayanrocks I don't think you can push to this PR. Would you please open a new one and I will close this?

Sure, Just assign the issue to me

**Comment by ashishbhate** [comment]
Closing, new PR at #7137

**Pull request closed without merging**